packages/router-plugin/src/core/router-code-splitter-plugin.ts

```diff
@@ -174,26 +159,13 @@ export const unpluginRouterCodeSplitterFactory: UnpluginFactory<
 
         userConfig = getConfig(options, ROOT)
       },
+      // handleHotUpdate({ file, server, modules }) {
+      //   return []
+      // },
     },
 
     rspack(compiler) {
       ROOT = process.cwd()
-
-      compiler.hooks.beforeCompile.tap(PLUGIN_NAME, (self) => {
-        self.normalModuleFactory.hooks.beforeResolve.tap(
-          PLUGIN_NAME,
-          (resolveData: { request: string }) => {
-            if (resolveData.request.includes(JoinedSplitPrefix)) {
-              resolveData.request = resolveData.request.replace(
-                JoinedSplitPrefix,
-                '',
-              )
-            }
-          },
-        )
-      })
-
-      userConfig = getConfig(options, ROOT)
```

Comment:
This also might be necessary still. Could use a second look by someone else.

Reply:
this should not be related at all for vite hmr, as this affects only rspack?

Reply:
You'll still need to keep this.
```ts
userConfig = getConfig(options, ROOT)
```

Reply:
done